lib/chef/resource.rb

```diff
@@ -1090,25 +1099,23 @@ def should_skip?(action)
     # === Returns
     # <Chef::Resource>:: returns the proper Chef::Resource class
     def self.resource_for_node(short_name, node)
-      require 'chef/resource_resolver'
       klass = Chef::ResourceResolver.new(node, short_name).resolve
       raise Chef::Exceptions::NoSuchResourceType.new(short_name, node) if klass.nil?
       klass
     end
 
+    #
     # Returns the class of a Chef::Resource based on the short name
+    # Only returns the *canonical* class with the given name, not the one that
+    # would be picked by the ResourceResolver.
+    #
     # ==== Parameters
     # short_name<Symbol>:: short_name of the resource (ie :directory)
     #
     # === Returns
     # <Chef::Resource>:: returns the proper Chef::Resource class
     def self.resource_matching_short_name(short_name)
-      begin
-        rname = convert_to_class_name(short_name.to_s)
-        Chef::Resource.const_get(rname)
-      rescue NameError
-        nil
-      end
+      Chef::ResourceResolver.new(Chef::Node.new, short_name).resolve
```

Comment:
And yeah,I added a super useful comment about what it actually does, and then changed what it did to something that made a bit more sense.  We should probably just leave it and deprecate.